### ISSUE TITLE
🐛`navigate = true` breaks `{commit,file}-style` settings

### ISSUE THREAD

**Issue by Kr1ss-XD** [dandavison/delta#256]
##### _Pls excuse me if this is a known issue already;_ at least I haven't found any ticket mentioning it in the title.

When delta's `--navigate` or `git config delta.navigate true` option is used, the `commit-style` and `file-style` config won't be respected (the commit descrptors and filenames won't be highlighted). The screenshot shows the output of `git show` _with_ and _without_ this option, on the left/right hand side, resp'ly :

![20200716-191341](https://user-images.githubusercontent.com/28707521/87708012-5ac96980-c7a2-11ea-9f8f-0ca5099ec763.png)

My first assumption was that this was likely related to the highlighting of the `commit` and `Δ` strings - or more precisely to the text highlighting being _reset_ at the end of these strings.

My `less` [environment](https://github.com/dandavison/delta/files/4933412/lessenv.txt) is not trivial, but the output of `LESS= LESSKEY= git show` does suffer from the issue, too.

**Comment by dandavison** [comment]
Hey @Kr1ss-XD, I don't think I can reproduce this. Let's try to find a minimal reproducible example. E.g., how does this behave for you?

```
git log -p | PAGER=less BAT_PAGER=less LESS= delta --no-gitconfig --navigate --side-by-side --file-style='bold blue "#dddddd"' --commit-style='bold red "#dddddd"'
```

That is working for me (MacOS, iTerm2)

<table><tr><td><img width=600px src="https://user-images.githubusercontent.com/52205/87734650-433db100-c7a1-11ea-8d9f-84bd422359e6.png" alt="image" /></td></tr></table>

**Comment by dandavison** [comment]
Can you try without any `LESS_TERMCAP_*` env vars?

Incidentally, seeing your config makes me realize that I need to do some reading up on what less is capable of! Do let me know if you have ever have thoughts about how delta could be making more use of less. (Did you happen to see #234 and #237 where we need to figure out how to make `navigate` behave better when the pattern isn't present, and when the output is less than one screen?)

**Comment by Kr1ss-XD** [comment]
This yields

![screen-20200717-0235](https://user-images.githubusercontent.com/28707521/87737132-7486a300-c7da-11ea-8058-8d2fe6c38297.png)

I'm using xfce4-terminal and tmux FWIW, but I get the same result outside of tmux and when using a different terminal emulator; I've tried xterm, urxvt and even a tty, so we can exclude that as the culprit.

Anyways, since `--navigate` _does not_ break the highlighting on your side, it's probably not a delta bug but some weird configuration issue on my end. I'll try to dig a bit deeper tomorrow and report back my findings, if any.

Thx so far for investigating.

**Comment by Kr1ss-XD** [comment]
:congratulations: :+1: You hit the nail by the head !  After unsetting `LESS_TERMCAP*`, I get
 
![screen-20200717-0322](https://user-images.githubusercontent.com/28707521/87738078-e06a0b00-c7dc-11ea-8ed6-8d97947a80e0.png)

> Do let me know if you have ever have thoughts about how delta could be making more use of less.

I might give you some thoughts regarding this one, but it's late here, sorry. I'll get back to you after work tomorrow. :)

**Comment by dandavison** [comment]
OK awesome. I shall read up on `LESS_TERMCAP` settings!

**Comment by dandavison** [comment]
Hey @Kr1ss-XD could you educate me on a point? What does this mean exactly?

> I've tried ... even a tty

**Comment by Kr1ss-XD** [comment]
Sure. That's a historic term, 'tty' abbreviates 'teletype', which was basically a printer and a keyboard in early UNIX days, later it meant a hardware terminal, which is a (dumb) keyboard+Text display unit.

On modern UNIX-like systems, you have several virtual consoles (besides the one that normally the display server runs on) which you can reach with `CTRL+F1` to `CTRL+Fn`, where n varies between distributions.

Basically, you can run text applications (most notably command shells) on tty's even when the graphical environment has crashed or fails to start up.

Not sure about Mac, but you could try e.g. `CTRL+F2` and if a tty opens on your box, you'll see what I mean ;)

**Comment by dandavison** [comment]
Thanks. The level of my understanding was that output from an application might be going to a "tty", versus e.g. a pipe, or a file/socket, but I did not know it was possible to test anything in a generic tty.  Does this suggest that what you're describing is not possible on MacOS? https://apple.stackexchange.com/questions/368044/do-terminal-ttys-for-ios-and-macos-exist#comment492618_368051

**Comment by Kr1ss-XD** [comment]
>output from an application might be going to a "tty", versus e.g. a pipe, or a file/socket,

Yeah, I guess it's validd to call this a tty too, in the context of an applications stdin/-out streams.

As for MacOS, I know nothing about this, but according to the article you linked it seems a Mac doesn't have Linux-like tty's.

**Comment by Kr1ss-XD** [comment]
Btw maybe you can simulate something like a tty environment for testing by setting
```
TERM=dumb
```
or
```
TERM=vt100
```
depending on how you terminal emulator handles this.

**Comment by dandavison** [comment]
Thanks for the instructions @Kr1ss-XD. I'll close this ticket since I think we basically got to the bottom of the main issue (but do re-open if there's a TERMCAP interaction that shouldn't exist.)
